frontend/src/components/pages/upload.vue

```diff
@@ -187,8 +194,23 @@
             },
 
             async stepSaveFileInfoForm(transitionNextStepAfterSave) {
-                await this.updateUpload(this.upload);
-                if(transitionNextStepAfterSave) { this.step = this.steps.step4UploadProgress; }
+                let valid = true;
+                if (this.upload.files){
+                    
+                    for (let i=0; i<this.upload.files.length; i++){
+                        if ((!this.upload.files[i].start_date) || (!this.upload.files[i].end_date)){
+                            this.errorAlert = true;
+                            this.errorText = "All start and end dates are required"
+                            valid = false;
+                        }
+                    }
+                }
+                if (valid){
+                    this.errorAlert = false;
+                    this.errorText = "";
+                    await this.updateUpload(this.upload);
+                    if(transitionNextStepAfterSave) { this.step = this.steps.step4UploadProgress; }
+                }
```

Comment:
Would err catching be needed here just in case?

Reply:
Agreed more error reporting is better, added.